tasks/configure.yml

```diff
@@ -41,77 +41,19 @@
   tags:
     - molecule-idempotence-notest
 
-- name: Restart express pm2 process
-  become: true
-  become_user: "{{ express_system_user }}"
-  command: "{{ express_pm2_path }} restart {{ express_service_name }}"
-  args:
-    chdir: "{{ express_app_path }}"
-  register: pm2_process
-  ignore_errors: true
-  changed_when: false
-
-- name: set express_service_exists fact
-  set_fact:
-    express_service_not_exists: "{{ pm2_process.rc != 0 }}"
-
-- name: set express_service_exists fact
-  set_fact:
-    express_service_exists: "{{ not express_service_not_exists }}"
-
-- name: creating pm2 process
-  block:
-    - name: Delete express pm2 process
-      become: true
-      become_user: "{{ express_system_user }}"
-      command: "{{ express_pm2_path }} delete all"
-      ignore_errors: true
-
-    - name: Start express pm2 app
-      become: true
-      become_user: "{{ express_system_user }}"
-      command: >
-        {{ express_pm2_path }} start dist/index.js
-        --name {{ express_service_name }}
-      args:
-        chdir: "{{ express_app_path }}"
-
-    - name: save the current PM2 running list
-      become: true
-      become_user: "{{ express_system_user }}"
-      command: "{{ express_pm2_path }} save"
-  when: express_service_not_exists
-
-- name: delete the old startup file
-  become: true
-  become_user: "root"
-  command: >
-    {{ express_pm2_path }} unstartup
-    -u {{ express_system_user }}
-    --service-name {{ express_service_name }}
-    --hp {{ express_system_user_home }}
-  ignore_errors: true
-  notify:
-    - start_express_service
-  tags:
-    - molecule-idempotence-notest  # noqa 301
-
-- name: Generate an express PM2 init service file
-  become: true
-  become_user: "root"
-  command: >
-    {{ express_pm2_path }} startup
-    -u {{ express_system_user }}
-    --hp {{ express_system_user_home }}
-    --service-name {{ express_service_name }}
-  notify:
-    - start_express_service
-  tags:
-    - molecule-idempotence-notest  # noqa 301
-
-- name: Cleanup pm2 processes  # HACK: this should not be happening
-  shell:
-    ps aux | grep PM2 | grep -v grep | awk '{print $2}' | xargs kill -9  # noqa 306
-  ignore_errors: true
-  tags:
-    - molecule-idempotence-notest  # noqa 301
+- name: pm2 configure
```

Comment:
Instead of doing this here, can we do it in the `meta`?  Here is are examples: [1](https://github.com/onaio/ansible-sqitch-migrations/blob/master/meta/main.yml#L19), [2](https://github.com/onaio/ansible-onadata/blob/master/meta/main.yml#L19).

Reply:
some of the envs that should be passed are outputs of havin run some tasks for example. pm2 start command needs to be passed the path where the index file is, the index file is an output of running build.
The problem is that I understand dependencies will run before the role actually executes.

Reply:
cool, makes sense